### PR TITLE
nsq_sketch: add option for finite number of iterations

### PR DESCRIPTION
**Pull request by tux0010**

### CODE REVIEW

File: nsq_sketch.go [hunk 1]
```diff
@@ -139,6 +142,12 @@ func (s *SketchHandler) TimerLoop(d time.Duration, r time.Duration) {
 			}
 			s.Unlock()
 		}
+
+		i++
+
+		if i == iterations {
```
Comment (jehiah) [inline nsq_sketch.go]: only `displayTicker` should count as an iteration.

**Review by jehiah — APPROVED** [review]
LGTM, just one small change

**Comment by tux0010** [comment]
Validated in the dev vm:

```
dev ~:$ ./nsq_sketch -topic=api_requests -key=api_path -lookupd-http-address=127.0.0.1:4161 -iterations=2
2017/05/18 13:55:49 Sketching for 2m0s intervals. Displaying 15 top items every 5s
2017/05/18 13:55:49 INF    1 [api_requests/sketch#ephemeral] querying nsqlookupd http://127.0.0.1:4161/lookup?topic=api_requests
2017/05/18 13:55:49 INF    1 [api_requests/sketch#ephemeral] (127.0.0.1:4150) connecting to nsqd

----- 5 messages ----
1 - /v4/users/:login
1 - /v4/organizations/:organizationGUID
1 - /v4/brands/:brandGUID
1 - /v4/organizations/:organizationGUID/bsds
1 - /v4/organizations/:organizationGUID/limits

----- 5 messages ----
1 - /v4/users/:login
1 - /v4/organizations/:organizationGUID
1 - /v4/brands/:brandGUID
1 - /v4/organizations/:organizationGUID/bsds
1 - /v4/organizations/:organizationGUID/limits
2017/05/18 13:55:59 INF    1 [api_requests/sketch#ephemeral] stopping...
2017/05/18 13:55:59 INF    1 [api_requests/sketch#ephemeral] (127.0.0.1:4150) received CLOSE_WAIT from nsqd
2017/05/18 13:55:59 INF    1 [api_requests/sketch#ephemeral] (127.0.0.1:4150) beginning close
2017/05/18 13:55:59 INF    1 [api_requests/sketch#ephemeral] (127.0.0.1:4150) readLoop exiting
2017/05/18 13:55:59 INF    1 [api_requests/sketch#ephemeral] (127.0.0.1:4150) breaking out of writeLoop
2017/05/18 13:55:59 INF    1 [api_requests/sketch#ephemeral] (127.0.0.1:4150) writeLoop exiting
2017/05/18 13:55:59 INF    1 [api_requests/sketch#ephemeral] (127.0.0.1:4150) finished draining, cleanup exiting
2017/05/18 13:55:59 INF    1 [api_requests/sketch#ephemeral] (127.0.0.1:4150) clean close complete
2017/05/18 13:55:59 WRN    1 [api_requests/sketch#ephemeral] there are 0 connections left alive
2017/05/18 13:55:59 INF    1 [api_requests/sketch#ephemeral] stopping handlers
2017/05/18 13:55:59 INF    1 [api_requests/sketch#ephemeral] exiting lookupdLoop
2017/05/18 13:55:59 INF    1 [api_requests/sketch#ephemeral] rdyLoop exiting
```

**Comment by tux0010** [comment]
Also validated infinite loop when iterations is not provided

**Comment by tux0010** [comment]
RFR/RFM /cc @jehiah

**Pull request merged**